SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Channel/Channel.cs

```diff
@@ -177,151 +164,126 @@ public Task Enqueue(Log log)
-                CheckPendingLogs();
+                using (await _mutex.GetLockAsync(state).ConfigureAwait(false))
+                {
+                    _enabled = true;
+                    _discardLogs = false;
+                    _mutex.InvalidateState();
+                    state = _mutex.State;
+                }
+                var logCount = await _storage.CountLogsAsync(Name).ConfigureAwait(false);
+                using (await _mutex.GetLockAsync(state).ConfigureAwait(false))
+                {
+                    _pendingLogCount = logCount;
+                }
             }
             catch (StatefulMutexException e)
             {
-                MobileCenterLog.Warn(MobileCenterLog.LogTag, "The CountFromDisk operation has been cancelled", e);
-            }
-            finally
-            {
-                _mutex.Unlock();
+                MobileCenterLog.Warn(MobileCenterLog.LogTag, "The Resume operation has been cancelled", e);
             }
+            await CheckPendingLogsAsync(state).ConfigureAwait(false);
         }
 
-        private Task Suspend(bool deleteLogs, Exception exception)
+        private async Task SuspendAsync(State state, bool deleteLogs, Exception exception)
         {
-            _enabled = false;
-            _batchScheduled = false;
-            _discardLogs = deleteLogs;
-            _stateKeeper.InvalidateState();
-            var stateSnapshot = _stateKeeper.GetStateSnapshot();
             try
             {
-                if (deleteLogs && FailedToSendLog != null)
-                {
-                    foreach (var log in _sendingBatches.Values.SelectMany(batch => batch))
-                    {
-                        _mutex.Unlock();
-                        FailedToSendLog?.Invoke(this, new FailedToSendLogEventArgs(log, exception));
-                        _mutex.Lock(stateSnapshot);
-                    }
-                }
-                try
-                {
-                    _ingestion.Close();
-                }
-                catch (IngestionException e)
+                using (await _mutex.GetLockAsync(state).ConfigureAwait(false))
                 {
-                    MobileCenterLog.Error(MobileCenterLog.LogTag, "Failed to close ingestion", e);
+                    _enabled = false;
+                    _batchScheduled = false;
+                    _discardLogs = deleteLogs;
+                    _mutex.InvalidateState();
                 }
-                if (deleteLogs)
+            }
+            catch (StatefulMutexException e)
+            {
+                MobileCenterLog.Warn(MobileCenterLog.LogTag, "The Suspend operation has been cancelled", e);
+            }
+            if (deleteLogs && FailedToSendLog != null)
+            {
+                foreach (var log in _sendingBatches.Values.SelectMany(batch => batch))
                 {
-                    _pendingLogCount = 0;
-                    return Task.Run(DeleteLogsOnSuspendedAsync);
```

Comment:
Now that this line is gone, `DeleteLogsOnSuspendedAsync` must be called. Consider adding this after the `if (deleteLogs && FailedToSendLog != null)` block:
```
            else if (deleteLogs)
            {
                await DeleteLogsOnSuspendedAsync();
            }
```